lms/djangoapps/grades/tests/integration/test_events.py

```diff
@@ -0,0 +1,126 @@
+"""
+Test grading event across apps.
+"""
+# pylint: disable=protected-access
+
+import json
+
+from capa.tests.response_xml_factory import MultipleChoiceResponseXMLFactory
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from lms.djangoapps.instructor.enrollment import reset_student_attempts
+from mock import patch
+from openedx.core.djangolib.testing.utils import get_mock_request
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from courseware.models import StudentModule
+from courseware.tests.test_submitting_problems import ProblemSubmissionTestMixin
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory
+from ...new.subsection_grade import SubsectionGradeFactory
+
+
+class DeleteStateEventIntegrationTest(ProblemSubmissionTestMixin, SharedModuleStoreTestCase):
+    """
+    Tests integration between the eventing in various layers
+    of the grading infrastructure.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(DeleteStateEventIntegrationTest, cls).setUpClass()
+        cls.course = CourseFactory.create()
```

Comment:
Can we create a split modulestore course instead?  You just need to wrap this line.
Then, you can validate that `course_version` is set correctly (and not-NULL) in `test_delete_student_state_events`.